clinica/iotools/utils/pipeline_handling.py

```diff
@@ -58,10 +60,114 @@ def t1_freesurfer_pipeline(caps_dir, df, freesurfer_atlas_selection=None, **kwar
     Returns:
          final_df: a DataFrame containing the information of the bids and the pipeline
     """
+
     from clinica.iotools.converters.adni_to_bids.adni_utils import (
         replace_sequence_chars,
     )
     from clinica.utils.stream import cprint
+    from pathlib import Path
+
+    # Ensures that df is correctly indexed
+    if "participant_id" in df.columns.values:
+        df.set_index(["participant_id", "session_id"], inplace=True, drop=True)
+
+    subjects_dir = Path(caps_dir) / "subjects"
+
+    pipeline_df = pd.DataFrame()
+
+    for participant_id, session_id in df.index.values:
+
+        ses_path = subjects_dir / participant_id / session_id
+        mod_path = ses_path / "t1"
+
+        try:
+            flag_skip = False
+            long_id = next(mod_path.glob("long*")).name
+        except StopIteration:
+            cprint(
+                f"Could not find a longitudinal dataset for participant {participant_id} {session_id}",
+                lvl="warning",
+            )
+            flag_skip = True
+
+        if flag_skip:
+            continue
+
+        mod_path = mod_path / long_id / "freesurfer_longitudinal" / "regional_measures"
+
+        ses_df = pd.DataFrame(
+            [[participant_id, session_id]], columns=["participant_id", "session_id"]
+        )
+        ses_df.set_index(["participant_id", "session_id"], inplace=True, drop=True)
+
+        if mod_path.exists():
+            # Looking for atlases
+            atlas_paths = sorted(
+                (mod_path).glob(f"{participant_id}_{session_id}_*volume.tsv")
+            )
+            for atlas_path in atlas_paths:
+                if "-wm_volume" not in str(atlas_path) and "-ba_volume" not in str(
+                    atlas_path
+                ):
+                    atlas_name = atlas_path.stem.split("_parcellation-")[1].split("_")[
+                        0
+                    ]
+                    if atlas_path.exists() and (
+                        not freesurfer_atlas_selection
+                        or (
+                            freesurfer_atlas_selection
+                            and atlas_name in freesurfer_atlas_selection
+                        )
+                    ):
+                        atlas_df = pd.read_csv(atlas_path, sep="\t")
+                        label_list = [
+                            "t1-freesurfer-longitudinal_atlas-" + atlas_name + "_" + x
+                            for x in atlas_df.label_name.values
+                        ]
+                        ses_df[label_list] = atlas_df["label_value"].to_numpy()
+
+            # Always retrieve subcortical volumes
+            atlas_path = next(
+                sorted(
+                    (mod_path).glob(
+                        f"{participant_id}_{session_id}_*segmentationVolumes.tsv"
+                    )
+                )
+            )
```

Comment:
What happens if we get a `StopIteration` exception here?

Reply:
Added try block to print warning and skip if applicable.